py-polars/polars/lazyframe/frame.py

```diff
@@ -1707,33 +1740,44 @@ def collect_async(
         comm_subplan_elim: bool = True,
         comm_subexpr_elim: bool = True,
         streaming: bool = False,
-    ) -> _AsyncDataFrameResult[DataFrame]:
+    ) -> Awaitable[DataFrame] | _GeventDataFrameResult[DataFrame]:
         """
         Collect DataFrame asynchronously in thread pool.
 
-        Collects into a DataFrame, like :func:`collect`
-        but instead of returning DataFrame directly its collected inside thread pool
-        and gets put into `queue` with `put_nowait` method,
+        Collects into a DataFrame, like :func:`collect` but instead of returning
```

Comment:
Maybe adjust punctuation: "Collects into a DataFrame (like :func:`collect`), but instead of returning dataframe directly, they are scheduled to be collected inside thread pool, ..."